agreements/notePurchaseAgreementsGateway.go

```diff
@@ -3,54 +3,57 @@ package agreements
 import (
 	"context"
 
+	"github.com/vireocloud/property-pros-service/data"
 	"github.com/vireocloud/property-pros-service/interfaces"
 )
 
 type NotePurchaseAgreementGateway struct {
-	repository interfaces.IAgreementsRepository
+	repository interfaces.IRepository[data.NotePurchaseAgreement]
 	factory    interfaces.INotePurchaseAgreementModelFactory
 }
 
-func NewNotePurchaseAgreementGateway(repository interfaces.IAgreementsRepository, factory interfaces.INotePurchaseAgreementModelFactory) interfaces.INotePurchaseAgreementGateway {
+func NewNotePurchaseAgreementGateway(repository interfaces.IRepository[data.NotePurchaseAgreement], factory interfaces.INotePurchaseAgreementModelFactory) *NotePurchaseAgreementGateway {
 	return &NotePurchaseAgreementGateway{
 		repository: repository,
 		factory:    factory,
 	}
 }
 
-func (g *NotePurchaseAgreementGateway) SaveNotePurchaseAgreement(ctx context.Context, model interfaces.IAgreementModel) (interfaces.IAgreementModel, error) {
-	result, err := g.repository.Save(model.GetPayload())
+func (g *NotePurchaseAgreementGateway) SaveNotePurchaseAgreement(ctx context.Context, agreement data.NotePurchaseAgreement) (data.NotePurchaseAgreement, error) {
+	_, err := g.repository.Save(&agreement)
 
 	if err != nil {
-		return nil, err
+		return data.NotePurchaseAgreement{}, err
 	}
 
-	return g.factory.NewPurchaseAgreementModel(ctx, result)
+	return agreement, nil
 }
 
 func (g *NotePurchaseAgreementGateway) Getall(ctx context.Context) ([]interfaces.IAgreementModel, error) {
-	results := g.repository.Query(nil)
-	models := []interfaces.IAgreementModel{}
+	return nil, nil
```

Comment:
The code is currently commented out, the new model mappings will need to be implemented if the proposal is accepted. The proposal is currently tested for saveNotePurchaseAgreement.